core/templates/pages/contributor-dashboard-page/modal-templates/certificate-download-modal.component.ts

```diff
@@ -104,7 +104,12 @@ export class CertificateDownloadModalComponent {
         this.toDate
       )
       .then((response: ContributorCertificateResponse) => {
-        this.createCertificate(response);
+        if (response) {
+          this.createCertificate(response);
+        } else {
```

Comment:
I believe you will have to add frontend tests for this as well.

Reply:
Frontend tests have been added! Thank you for the feedback!